packages/shared/src/vault/executeSwap.ts

```diff
@@ -106,43 +101,38 @@ const callNative = async (
     params.ccmMetadata.message,
     params.ccmMetadata.gasBudget,
     [],
-    { value: params.amount, ...opts },
+    { value: params.amount, ...extractOverrides(txOpts) },
   );
 
-  return transaction.wait(1);
+  return transaction.wait(txOpts.wait);
 };
 
 const callToken = async (
   params: TokenCallParams,
-  {
-    signer,
-    network,
-    vaultContractAddress: vaultAddress,
-    srcTokenContractAddress: tokenAddress,
-    ...opts
-  }: ExecuteOptions,
+  opts: SwapNetworkOptions,
```

Comment:
i think i would rename opts to `networkOpts` everywhere we have two types of opts for the sake of clarity

Reply:
sounds good!